packages/styled-components/src/types.ts

```diff
@@ -67,13 +65,16 @@ export type Interpolation<Props> =
   | IStyledComponent<'web', any, any>
   | Interpolation<Props>[];
 
-export type Attrs<Props> =
-  | (ExtensibleObject & Props)
+export type Attrs<Props extends {}> =
+  | (ExecutionProps & Props)
```

Comment:
I think this should be:
```suggestion
  | Partial<ExecutionProps & Props>
```

Reply:
@joshjg You're probably right, but this stuff can be a doozy to think about. Would you be able to provide an example of before/after behavior to aid my understanding?